var/spack/repos/builtin/packages/nseg/package.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import os
+
+
+class Nseg(MakefilePackage):
+    """NSEG - Low complexity sequence identification
+
+       This package is hosted as an FTP directory. To install it, you will
+       need to download all of the files from the 'homepage' and tar them."""
```

Comment:
I think the checksum will be dependent on the version of tar used to create the tarball, correct? There has to be a better way to do this. Maybe reach out to the developers and see if there is an existing tarball somewhere? I had this problem with another package and I found a mirror in GitHub that someone made.

Reply:
I couldn't find any tarballs hosted officially or on any github mirrors. There are only 7 files in the source code - I thought about adding each one individually as a resource and checksumming each file, would that be better?

Reply:
@tgamblin @scheibelp any thoughts on how to handle this? I had the exact same problem with [STREAM](http://www.cs.virginia.edu/stream/FTP/Code/), which only distributes individual files in FTP. Luckily I found someone who made a GitHub repo that clones it. Perhaps you could do this yourself as a workaround?

Reply:
I like the idea of pointing to a single file in the ftp dir as the package, and treating the rest like resources.  Not sure if there's another way to get a trusted version in spack.  There doesn't seem to be any license info, so unclear whether we can redistribe on a github repo.

Reply:
I went ahead and implemented this idea -- it seems to work OK and properly checksums all of the files. Things could get messy if the package is ever updated but that hasn't happened in 19 years.

Reply:
Defining a bunch of resources would work although it would be onerous. Another option I haven't thought through 100% is defining a new `FetchStrategy` and overriding `Nseg.fetcher`:

```
@property
def fetcher(self):
    return CustomFetcher(ftp_url)
```

That will involve defining a new fetcher that inherits from `FetchStrategy`. The most important methods to implement are `fetch`, `check`, and `expand`. You can arrange for everything to be downloaded into a single directory you create under `self.stage.path` in `fetch` (in which case `expand` can also be a noop).

Reply:
I'm fine with this current approach if you just want to get things merged. I still think we should find an easier way to do this in the long term though, but that shouldn't hold up this PR. So far `stream` and `nseg` are the only packages I know of that need this.

Reply:
@scheibelp I think some kind of `FTPDirectory` FetchStrategy would be great, and at least a couple packages would benefit it. But I have no idea how much work that is.